src/baldr/edgeinfo.cc

```diff
@@ -150,6 +202,64 @@ std::vector<std::pair<std::string, uint8_t>> EdgeInfo::GetTaggedNamesAndTypes()
   return name_type_pairs;
 }
 
+std::unordered_map<uint8_t, std::pair<uint8_t, std::string>> EdgeInfo::GetPronunciationsMap() const {
+  std::unordered_map<uint8_t, std::pair<uint8_t, std::string>> key_type_value_pairs;
+  key_type_value_pairs.reserve(name_count());
+  const NameInfo* ni = name_info_list_;
+  for (uint32_t i = 0; i < name_count(); i++, ni++) {
+    if (!ni->tagged_)
+      continue;
+
+    if (ni->name_offset_ < names_list_length_) {
+      std::string name = names_list_ + ni->name_offset_;
+      if (name.size() > 1) {
+        uint8_t num = 0;
+        try {
+          num = std::stoi(name.substr(0, 1));
```

Comment:
this is really strange. if its tagged why cant we just use the byte directly as an integer? surely we arent storing the values as strings!? if so that was a huge  mistake.

i expected the code to be:

`auto tag_type = static_cast<uint8_t>(name[0])`